sky/backends/backend_utils.py

```diff
@@ -859,6 +859,19 @@ def write_cluster_config(
     if isinstance(cloud, clouds.GCP):
         gcp_project_id = cloud.get_project_id(dryrun=dryrun)
 
+    specific_reservations = set(
+        skypilot_config.get_nested(('gcp', 'specific_reservations'), set()))
+
+    reservations = to_provision.get_reservations_available_resources(
+        specific_reservations)
+
+    filtered_specific_reservations = [
+        r for r, available_resources in reservations.items()
+        if r in specific_reservations and available_resources > 0
+    ]
+    num_reserved_workers = max(
+        min(sum(reservations.values()) - 1, num_nodes - 1), 0)
```

Comment:
It seems the `num_reserved_workers` should actually be `num_specific_reserved_workers`. Reason: we should not set the `specific_reservation` for the workers, when we want to have it use the automatic reservation, so the count should be the total number of specific reservations, instead of all the reservations.

Before Google adds the new logic, we probably want to use the following code instead. (We can also rename the `num_reserved_workers` to `num_specific_reserved_workers`)

```suggestion
    available_specific_reservations = sum(
        available_resources for r, available_resources in reservations.items()
        if r in specific_reservations)
    num_reserved_workers = max(
        min(available_specific_reservations - 1, num_nodes - 1), 0)

```